pkg/controller/provider/container/ovirt/client.go

```diff
@@ -33,14 +33,34 @@ type Client struct {
 	// Raw client.
 	client *libweb.Client
 	// Secret.
-	secret *core.Secret
+	secret                *core.Secret
+	clientExpiration      time.Time
+	clientTimeout         time.Duration
+	accessTokenExpiration time.Time
+	log                   logr.Logger
+}
+
+type ovirtTokenResponse struct {
+	AccessToken string `json:"access_token"`
+	Expiration  string `json:"exp"`
 }
 
 //
 // Connect.
 func (r *Client) connect() (err error) {
 	var TLSClientConfig *tls.Config
 
+	if time.Now().After(r.clientExpiration) {
```

Comment:
we can check here if client != nil to avoid printing the log about timeout exceeded / token expired the first time we connect

Reply:
Added `!isZero` check, it should only be zero when the client was connected for the first time

Reply:
ok, also works :)